### PR TITLE
core/tracing: add verkle gas change reasons to changelog

### PR DESCRIPTION
**Pull request by s1na**
Add changes from #30409 and #29338 to changelog.

Btw @gballet seems like `GasChangeWitnessContractCreation` is unused.

### CODE REVIEW

**Review by holiman — APPROVED** [review]
LGTM

File: core/tracing/CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,24 @@
...
+- `GasChangeReason` has been extended with the following reasons which will be enabled only post-Verkle. There shouldn't be any gas changes with those reasons prior to the fork.
+  - `GasChangeWitnessContractCollisionCheck` flags the event of adding to the witness when checking for contract address collision.
+
+## [v1.14.4]
+
+This release contained only minor extensions to the tracing interface.
+
+### Modified types
+
+- `GasChangeReason` has been extended with the following reasons will be active only post-Verkle.
```
Comment (gballet) [inline core/tracing/CHANGELOG.md:18]:
```suggestion
- `GasChangeReason` has been extended with the following reasons that will only be active post-Verkle.
```

File: core/tracing/CHANGELOG.md [hunk 2]
```diff
@@ -2,6 +2,24 @@
...
+  - `GasChangeWitnessContractInit` flags the event of adding to the witness during the contract creation initialization step.
+  - `GasChangeWitnessContractCreation` flags the event of adding to the witness during the contract creation finalization step.
+  - `GasChangeWitnessCodeChunk` flags the event of adding one or more contract code chunks to the witness.
+
```
Comment (gballet) [inline core/tracing/CHANGELOG.md:22]: it's missing the collision check

```suggestion
  - `GasChangeWitnessContractCollisionCheck` flags the event of adding to the witness when checking for contract address collision.
```

Comment (s1na) [reply core/tracing/CHANGELOG.md:22]: That goes in the "unreleased" section

**Comment by gballet** [comment]
> Btw @gballet seems like GasChangeWitnessContractCreation is unused.

It's going to be, it's just a matter of merging more verkle code.

**Pull request merged**